src/main/java/it/smartcommunitylab/aac/spid/provider/SpidIdentityProvider.java

```diff
@@ -120,14 +123,30 @@ public SubjectResolver<SamlUserAccount> getSubjectResolver() {
 
     @Override
     public String getAuthenticationUrl() {
-        // TODO build a realm-bound url, need updates on filters
-        return "/auth/" + getAuthority() + "/authenticate/";
+        return getLoginUrl();
+    }
+
+    @Override
+    public String getAuthenticationUrl(HttpServletRequest request) {
+        String registrationId = null;
+        if (request.getParameter(UPSTREAM_IDP_PARAMETER_NAME) != null) {
```

Comment:
the parameter should be plaintext. The logic for addressing the composite identifier is an internal one, we don't need to expose it to clients.

A request should look like
```
...authorize?idp_hint=PROVIDER_ID&domain_hint=https://spid.test
```

and then inside the method we can apply the encoding required to build the redirect.